tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/contact/EmailAddressContact.scala

```diff
@@ -82,6 +86,14 @@ class InMemoryEmailAddressContactSearchEngine extends EmailAddressContactSearchE
     SMono.fromCallable(() => domainList.put(domain, addressContact))
       .`then`(SMono.just(addressContact))
 
+  override def update(accountId: AccountId, mailAddress: MailAddress, updatedFields: ContactFields): Publisher[EmailAddressContact] =
+    SMono.fromPublisher(delete(accountId, mailAddress))
+      .`then`(SMono.fromPublisher(index(accountId, updatedFields)))
```

Comment:
If we use Guava Table here, I guess we can optimize this by directly putting data instead of doing a delete before adding.

Reply:
Do we really care of having a super optimized mem version though?

Reply:
At the end... no. For the case that if the email address changes. It's better to just delete the old record and putting the new one. If we directly putting data it can create inconsistencies on the mail address if that one changes (and this case is valid)